src/olympia/addons/views.py

```diff
@@ -658,7 +658,11 @@ class AddonViewSet(RetrieveModelMixin, GenericViewSet):
               AllowReviewer, AllowReviewerUnlisted),
     ]
     serializer_class = AddonSerializer
-    addon_id_pattern = re.compile(r'^(\{.*\}|.*@.*)$')
+    addon_id_pattern = re.compile(
+        # Match {uuid} or something@host.tld ("something" being optional)
+        # guids. Copied from mozilla-central XPIProvider.jsm.
+        r'^(\{[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}\}'
+        r'|[a-z0-9-\._]*\@[a-z0-9-\._]+)$')
```

Comment:
The regex definition in XPIProvider.jsm ignores case.  Should the same be done here?